.github/workflows/lint-backend.yml

```diff
@@ -1,6 +1,7 @@
 name: Python Linting with Ruff
 
-on: push
```

Comment:
How about we have
```
  push:
    branches:
      - main
```

so that the main branch gets CI runs in addition to pull requests?
My opinion might be biased, but running CI on the main branch every time we push commits (such as merging pull requests) can indicate that the default branch has no issues